Improved climate component behaviour

KPWhiver opened the pull request:
The hot water climate component now has three modes:
OFF: Hot water blocking will be requested
AUTO: Water will be heated when the tap opens
HEAT: Water will always be kept hot

The heating circuits climate components now also have an AUTO mode. When put in AUTO mode, they will relinquish control to the thermostat.

The room climate component now also has an AUTO mode. When put in AUTO mode, it will allow the thermostat to override the target temperature whenever it wants to (for example, because it is on a schedule).

KPWhiver committed:
Add button that triggers a "hot water push"

KPWhiver committed:
Set the hot water temperature if the heater does not supply it

KPWhiver committed:
Make the hot water and heater behaviour more logical

All climates now have an AUTO mode, which essentially means that the
thermostat is in control

KPWhiver committed:
Fix compile error

KPWhiver committed:
Determine mode of hot water based on master state

KPWhiver committed:
Set room thermostat climate component to AUTO when a manual override occurs

M66B commented:
It looks like blocking hot water doesn't work for me, unfortunately :-(

I think this is not correct ("Verwarmen") because the room temperature is higher than the set temperature:

![image](https://github.com/user-attachments/assets/194ea5ca-3c86-4d65-b0d3-6b72f6b1de25)

M66B commented:
'flame' is not on in any case.

I've gone through your commits, and I don't see anything suspicious, only nice code ;-)
I'm not fluent in climate components, though. What will be relevant (later) is this:

https://developers.esphome.io/blog/2025/05/14/_schema-deprecations/

You can contact me via marcel [apestaartje] bokhorst [punt] eu if you wish. Perhaps that's easier.

M66B commented:
After switching back to the main branch:

![image](https://github.com/user-attachments/assets/edea50b7-61ed-417d-ab15-f48dcb17a6dc)

KPWhiver commented:
(Sorry for the late response, was gone for the weekend.) I'm having trouble replicating this. What mode is the thermostat in, auto or heat? If you can give me your current config, I can see if I can replicate it here.

Flame is not necessary for the thermostat to be "heating". It just means that a thermostat (or the gateway) told the heater to heat, the heater does not need to listen (this could be the case when tap water is also being used).

About the tap water, are you at least able to set the tap water temperature? If so you could use that as a way to block the heater from heating the tap water. 

Thanks for the schema deprecation notice, I'll have a look.

M66B commented:
I hope you had a nice weekend, there is no need to hurry for anything. I'm just glad I could replace the OTGW default solution with an esphome solution, so I can manage everything from the same place, the esphome builder. Moreover, esphome is a rather stable foundation, and in the long run, non-custom solutions are mostly better. Given the rapidly growing number of home assistant / esphome users, I'm considering home assistant as non-custom now.

The repository wasn't easy to find, but I guess more people would be interested in switching if it would be more visible.

I had this problem after a factory reset too. The yaml config is here:

https://gist.github.com/M66B/6a87cb1bbbfc05154c46ba7e1a7e6fa1

I use the master branch with the pull requested patched in. If you want, I can do that again, and share logging, etc.

M66B commented:
Since I guess that could matter, I'm using an old/classic Honewell thermostat and an Intergas Xtreme 30 CW4 heater.

M66B commented:
Master = this pull requested patched in (another try):

![image](https://github.com/user-attachments/assets/822ad615-c9f5-4f4c-bf27-192c342ed2fb)

M66B commented:
It is heating as well:

![image](https://github.com/user-attachments/assets/294d57c6-7eb4-4901-899d-07c46b9bda1f)

![image](https://github.com/user-attachments/assets/480e2195-bd7d-4cf4-911d-6eb7ed9c4488)

Probably hot water.

M66B commented:
Using gas as well:

![image](https://github.com/user-attachments/assets/0e055f23-5fdf-45c0-a4bb-9d88a2dff32a)

KPWhiver commented:
I've been able to reproduce it, let's see if I can find the issue.

KPWhiver committed:
Fix an issue where the temperature was refreshed but the mode wasn't

This caused the temperature to be set to 5 degrees and the heater to
start heating.

KPWhiver commented:
Issue is fixed. Let me know if it works for you as well. I should really start looking into esphome automated testing.

M66B commented:
It works for me too. Nice!

Suggestion: link to this project with a comment here:

https://gathering.tweakers.net/forum/list_messages/1653967

And perhaps contact the topic starter for a link in the opening post.

I'm sure at least some people will be interested.

KPWhiver committed:
Remove override_thermostat option as it is has been replaced by AUTO climate mode

KPWhiver commented:
> It works for me too. Nice!
> 
> Suggestion: link to this project with a comment here:
> 
> https://gathering.tweakers.net/forum/list_messages/1653967
> 
> And perhaps contact the topic starter for a link in the opening post.
> 
> I'm sure at least some people will be interested.

By the way, thanks for the suggestion, will do!
